sql/core/benchmarks/DataSourceReadBenchmark-results.txt

```diff
@@ -2,430 +2,459 @@
+Data column - Parquet Vectorized: DataPageV1                 171            176           8         92.0          10.9     126.8X
+Data column - Parquet Vectorized: DataPageV2                 413            416           5         38.1          26.2      52.5X
+Data column - Parquet MR: DataPageV1                        2501           2511          13          6.3         159.0       8.7X
+Data column - Parquet MR: DataPageV2                        2356           2370          20          6.7         149.8       9.2X
+Data column - ORC Vectorized                                 279            283           4         56.3          17.8      77.6X
+Data column - ORC MR                                        2012           2016           6          7.8         127.9      10.8X
+Partition column - CSV                                      6557           6595          53          2.4         416.9       3.3X
+Partition column - Json                                     9875           9890          21          1.6         627.8       2.2X
+Partition column - Parquet Vectorized: DataPageV1             38             43          10        410.9           2.4     566.4X
+Partition column - Parquet Vectorized: DataPageV2             38             40           7        417.4           2.4     575.3X
+Partition column - Parquet MR: DataPageV1                   1182           1190          10         13.3          75.2      18.3X
+Partition column - Parquet MR: DataPageV2                   1173           1173           0         13.4          74.6      18.5X
+Partition column - ORC Vectorized                             44             47           8        356.8           2.8     491.7X
+Partition column - ORC MR                                   1267           1267           1         12.4          80.5      17.1X
+Both columns - CSV                                         23152          23178          37          0.7        1471.9       0.9X
+Both columns - Json                                        15665          15673          11          1.0         995.9       1.4X
+Both columns - Parquet Vectorized: DataPageV1                196            202           6         80.1          12.5     110.4X
+Both columns - Parquet Vectorized: DataPageV2                440            444           4         35.7          28.0      49.2X
+Both columns - Parquet MR: DataPageV1                       2538           2541           4          6.2         161.4       8.5X
+Both columns - Parquet MR: DataPageV2                       2248           2258          14          7.0         142.9       9.6X
+Both columns - ORC Vectorized                                314            319           8         50.1          20.0      69.1X
+Both columns - ORC MR                                       2092           2100          12          7.5         133.0      10.4X
+
+
+================================================================================================
```

Comment:
The result of `Vectorized Scan Multiple Partition Columns` without this pr as follows:
```
================================================================================================
Vectorized Scan Multiple Partition Columns
================================================================================================

OpenJDK 64-Bit Server VM 1.8.0_332-b09 on Linux 5.13.0-1023-azure
Intel(R) Xeon(R) Platinum 8370C CPU @ 2.80GHz
Vectorized Scan 1 partition columns:      Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
------------------------------------------------------------------------------------------------------------------------
Parquet Vectorized: DataPageV1                      420            427           8         37.5          26.7       1.0X
Parquet Vectorized: DataPageV2                      418            434          16         37.6          26.6       1.0X
ORC Vectorized                                      411            428          14         38.2          26.2       1.0X

OpenJDK 64-Bit Server VM 1.8.0_332-b09 on Linux 5.13.0-1023-azure
Intel(R) Xeon(R) Platinum 8370C CPU @ 2.80GHz
Vectorized Scan 2 partition columns:      Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
------------------------------------------------------------------------------------------------------------------------
Parquet Vectorized: DataPageV1                      447            452           5         35.2          28.4       1.0X
Parquet Vectorized: DataPageV2                      452            468          20         34.8          28.8       1.0X
ORC Vectorized                                      484            488           3         32.5          30.7       0.9X

OpenJDK 64-Bit Server VM 1.8.0_332-b09 on Linux 5.13.0-1023-azure
Intel(R) Xeon(R) Platinum 8370C CPU @ 2.80GHz
Vectorized Scan 3 partition columns:      Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
------------------------------------------------------------------------------------------------------------------------
Parquet Vectorized: DataPageV1                      524            531           7         30.0          33.3       1.0X
Parquet Vectorized: DataPageV2                      530            533           3         29.7          33.7       1.0X
ORC Vectorized                                      547            556          11         28.8          34.8       1.0X
```

Reply:
with this pr:

```
================================================================================================
Vectorized Scan Multiple Partition Columns
================================================================================================

OpenJDK 64-Bit Server VM 1.8.0_332-b09 on Linux 5.13.0-1023-azure
Intel(R) Xeon(R) Platinum 8370C CPU @ 2.80GHz
Vectorized Scan 1 partition columns:      Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
------------------------------------------------------------------------------------------------------------------------
Parquet Vectorized: DataPageV1                      411            423          19         38.3          26.1       1.0X
Parquet Vectorized: DataPageV2                      412            416           5         38.2          26.2       1.0X
ORC Vectorized                                      523            527           3         30.1          33.2       0.8X

OpenJDK 64-Bit Server VM 1.8.0_332-b09 on Linux 5.13.0-1023-azure
Intel(R) Xeon(R) Platinum 8370C CPU @ 2.80GHz
Vectorized Scan 2 partition columns:      Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
------------------------------------------------------------------------------------------------------------------------
Parquet Vectorized: DataPageV1                      417            422           4         37.7          26.5       1.0X
Parquet Vectorized: DataPageV2                      418            426           8         37.7          26.5       1.0X
ORC Vectorized                                      483            487           6         32.6          30.7       0.9X

OpenJDK 64-Bit Server VM 1.8.0_332-b09 on Linux 5.13.0-1023-azure
Intel(R) Xeon(R) Platinum 8370C CPU @ 2.80GHz
Vectorized Scan 3 partition columns:      Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
------------------------------------------------------------------------------------------------------------------------
Parquet Vectorized: DataPageV1                      428            437          13         36.8          27.2       1.0X
Parquet Vectorized: DataPageV2                      427            430           3         36.8          27.2       1.0X
ORC Vectorized                                      543            569          25         29.0          34.5       0.8X
```